packages/protocol-utils/src/orders.ts

```diff
@@ -129,6 +135,35 @@ export abstract class OrderBase {
         }
     }
 
+    /*
+     * This is an advanced order signature function that should only be used if you know what you're doing.
+     * It allows you to bring your own wallet signature callback function, this can be used to sign orders
+     * with a wallet that is otherwise unsupported by Web3Wrapper.
+     * The signature function needs to implement correct signing of the data and return the results as a string.
+     */
+    public async _getSignatureWithSignFunctionAsync(
```

Comment:
I don't really get the value this function adds. Why not just export `parseRpcSignature()`? Sure we call either `getEIP712TypedData()` or `getHash()` based on `type`, but it seems like the caller already has to know in advance which one they're working with because they have to pass in the function that accepts it. Or is the prescription that the callback should handle both cases? In which case shouldn't it be `type SignatureCallbackFunction = (dataOrHash: EIP712TypedData | string, signer: string) => Promise<string>;`?

Reply:
It doesn't really add much value tbh! I was initially gonna do a big refactor and create a minimal dependency file for this logic but then I got lazy and appreciative of the convenience of the `LimitOrder` object. 

But yeah, maybe I should give that another stab and realize the original vision 🤓  We are getting so rekt on bundle size because we don't use `import type {} from '@0x/...` when we only need the types 🙈 

I'll fix this tomorrow 👍